src/cli/path_validation.py

```
@@ -0,0 +1,144 @@
+    Raises:
+        typer.BadParameter: When the pair is incoherent by any of the
+            three rules above. The message names the specific violation
+            so the user can spot the argument ordering mistake.
+    """
+    in_resolved = _resolve_user_path(input_dir)
+    out_resolved = _resolve_user_path(output_dir)
+
+    if in_resolved == out_resolved:
+        raise typer.BadParameter(
+            f"Input and output refer to the same path: {in_resolved!s}. Pass different directories."
+        )
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)